store/root.js

```diff
@@ -100,54 +107,51 @@ class Root extends Store {
     this.organisation.retrieve()
   }
 
-  @action async changeDataProvider(id) {
+  @action changeDataProvider(id) {
     // Clean-up or initial request
     if (id == null) {
       this.dataProvider = null
       return
     }
 
-    const dataProviderInt = parseInt(id, 10)
-
     // Probably a repeated request. No need to change
-    // Compare strings and numbers
-    // since we do not really know what type of ID the API exposes
-    if (this.dataProvider?.id === dataProviderInt) return
-
-    let dataProviderMetaData = this.user.getDataProviderById(dataProviderInt)
+    //
+    // Compare strings and numbers since we do not know what type of ID
+    // the API exposes
+    // eslint-disable-next-line eqeqeq
+    if (this.dataProvider?.id == id) return
 
     // Check access rights
-    if (
-      dataProviderMetaData !== null &&
-      !this.user.canManage(dataProviderMetaData?.id)
-    ) {
-      if (!this.user.superAdmin) {
-        const dpStr = `DataProvider#${id}`
-        throw new AccessError(
-          `${this.user} does not have access to the ${dpStr}`
-        )
-      }
-
-      try {
-        const { data } = await this.user.retrieveDataProviderMetadata(id)
-        dataProviderMetaData = data
-        this.user.dataProviders = [
-          ...this.user.dataProviders,
-          dataProviderMetaData,
-        ]
-      } catch (error) {
-        throw new AccessError(
-          `Failed to load DataProvider#${id} metadata. Server responded with ${error.status}`
-        )
-      }
+    if (!this.user.canManage(id)) {
```

Comment:
If you allow all repositories for super admin (even the non-existing ones) you will never get error reporting. 

If you want to see 404 page you need to throw an error here hence I was fetching repository metadata in this method and thrown exception if it returned any status code but 200.

Reply:
Non-existing data provider will get infinite loading or something. It will try to solve it but I think it's not bad as it is.

Reply:
The current implementation redirects to the first existing data provider and I would say it should behave the same way.

Reply:
You will not like the implementation (d520f7c), I guess :) Let me know, what you think